pkg/crc/api/client_test.go

```diff
@@ -0,0 +1,55 @@
+package api
+
+import (
+	"errors"
+
+	"github.com/code-ready/crc/pkg/crc/machine"
+	"github.com/code-ready/crc/pkg/crc/network"
+)
+
+type mockClient struct {
+}
+
+func (c mockClient) Delete(deleteConfig machine.DeleteConfig) (machine.DeleteResult, error) {
+	return machine.DeleteResult{}, errors.New("not implemented")
+}
```

Comment:
This almost duplicates what is in `cmd/crc/cmd/client_test.go`, does it have to be this way?

Reply:
done :+1: 

Introduced a fakemachine package with this Client.